rust/operator-binary/src/airflow_controller.rs

```diff
@@ -518,17 +539,56 @@ pub async fn reconcile_airflow(
                 })?;
         }
 
+        // if the replicas are changed at the same time as the reconciliation
+        // being paused, it may be possible to have listeners that are *expected*
+        // (according to their replica number) but which are not yet created, so
+        // deactivate this action in such cases.
+        if airflow.spec.cluster_operation.reconciliation_paused
+            || airflow.spec.cluster_operation.stopped
+        {
+            tracing::info!(
+                "Cluster is in a transitional state so do not attempt to collect listener information that will only be active once cluster has returned to a non-transitional state."
```

Comment:
I would mention what is actually happening, either no reconcilation or the cluster is stopped?

Reply:
5c7022c